src/components/Sorting.vue

```diff
@@ -0,0 +1,114 @@
+<template>
+  <div class="flex w-full justify-between flex-wrap gap-5">
+    <div class="flex gap-3">
+      <button
+        @click="handleClick('title')"
+        :class="{
+          'sort-buttons': true,
+          ascendant: sortByTitle && sortDirection === 'asc',
+          descendant: sortByTitle && sortDirection !== 'asc',
+        }"
+      >
+        Title
+      </button>
+      <button
+        @click="handleClick('text')"
+        :class="{
+          'sort-buttons': true,
+          ascendant: sortByText && sortDirection === 'asc',
+          descendant: sortByText && sortDirection !== 'asc',
+        }"
+      >
+        Description
+      </button>
+      <button
+        @click="handleClick('priority')"
+        :class="{
+          'sort-buttons': true,
+          ascendant: sortByPriority && sortDirection === 'asc',
+          descendant: sortByPriority && sortDirection !== 'asc',
+        }"
+      >
+        Priority
+      </button>
+      <button
+        @click="handleClick('date')"
+        :class="{
+          'sort-buttons': true,
+          ascendant: sortByDate && sortDirection === 'asc',
+          descendant: sortByDate && sortDirection !== 'asc',
+        }"
+      >
+        Date
+      </button>
+    </div>
+
+    <div class="flex gap-2">
+      <button
+        @click="handleSortDirectionChange('asc')"
+        class="ascend-descend-btn bg-emerald-400"
+      >
+        <img src="../assets/Ascending.svg" />
+      </button>
+      <button
+        @click="handleSortDirectionChange('desc')"
+        class="ascend-descend-btn bg-black"
+      >
+        <img src="../assets/Descending.svg" />
+      </button>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref } from 'vue';
+import { Todotype } from '../Types/toDo';
+import { SortDirection } from '../Types/SortDirection';
+
+const sortBy = ref('');
+const sortByTitle = ref(false);
```

Comment:
As I mentioned before, the boolean values have to contain the `is` or `has` prefix. Please review all of the boolean variables and rename them.  ex. `isSortedByText`